src/libsync/propagateupload.cpp

```diff
@@ -676,6 +676,44 @@ void PropagateUploadFileCommon::commonErrorHandling(AbstractNetworkJob *job)
     SyncFileItem::Status status = classifyError(job->reply()->error(), _item->_httpErrorCode,
         &propagator()->_anotherSyncNeeded, replyContent);
 
+    // Remote host closed the connection after an upload larger than _failsafeMaxChunkSize
+    if (job->reply()->error() == QNetworkReply::RemoteHostClosedError 
+        && _item->_requestBodySize > propagator()->syncOptions()._failsafeMaxChunkSize) {
+        // treat it as if the server returned HTTP-413
+        qCInfo(lcPropagateUpload()) << "Remote host closed the connection when uploading" 
+                                    << Utility::octetsToString(_item->_requestBodySize) << "."
+                                    << "Treating the error like HTTP-413 (request entity to large)";
+
+        _item->_httpErrorCode = 413;
+    }
+
+    // Request entity to large: Chunk size is larger than supported by the server
+    if (_item->_httpErrorCode == 413 && _item->_requestBodySize > 0 && status == SyncFileItem::NormalError) {
+        auto opts = propagator()->syncOptions();
+        auto requestSize = _item->_requestBodySize;
+
+        // Set new bound as account upload limit (we need to guess here, as sent bytes are no good indicator what is allowed)
+        auto maxRequestSize = requestSize > opts._failsafeMaxChunkSize
+            ? opts._failsafeMaxChunkSize
+            : qMax(opts._minChunkSize, requestSize / 2);
+        
+        propagator()->account()->setMaxRequestSizeIfLower(maxRequestSize);
```

Comment:
Not sure if I am missing something, but, will we do this calculation and set it each time we restart the desktop client? In this case, it makes sense to store the calculated value (`calculatedRequestSize`?) in the config per account and use it for future communication with the server from the current account even after restarting the app. This will also help to avoid having 413 errors displayed in UI after restarting the app.

Reply:
It actually was on purpose to let the NC client retry when the app is restarted.

The idea behind this is that the web server config might be changed or might depend on the location, e.g. a WAF with stronger limits when accessed via Internet vs VPN or local network.

It may even make sense to set the max request size to 0 e.g. once a day or when the network changes. But this is optional, the main point was that it shouldn’t be persistent.

Reply:
Btw, regarding UI, the error is displayed only as long as the upload didn’t succeed. After that it is removed.

Reply:
Please see latest updates d93c76825b379c55aee210c11d6af29450b924b4. 
Auto-sensed size is persisted now, but the agent still tries configured limits once after a restart.